tests/dragonfly/replication_test.py

```diff
@@ -1068,3 +1067,107 @@ async def test_readonly_script(df_local_factory):
         assert False
     except aioredis.ResponseError as roe:
         assert 'READONLY ' in str(roe)
+
+
```

Comment:
how about adding a test for a very small timeout that the replica will not be able to takeover?

Reply:
also a test that replica failes during the takeover and master is failing during failover

Reply:
lets add this test in a different PR